### ISSUE TITLE
oneplus2: Unable to unlock bootloader for flashing

### ISSUE THREAD

**Issue by Kreyren** [Divested-Mobile/DivestOS-Build#6]
Since the installation of DivestOS i am unable to get `fastboot flash ...` to work:

```console
# While device is in bootloader (fastboot mode)
kreyren@leonid:~/Downloads$ fastboot flash recovery recovery.img
Sending 'recovery' (22180 KB)                      OKAY [  0.762s]
Writing 'recovery'                                 FAILED (remote: 'device is locked. Cannot flash images')
fastboot: error: Command failed
```

I've enabled `Android debugging` and `OEM unlocking` in `Developer settings` and approved the system in popup for `USB debugging` and i am getting a notification about `USB debugging connected`

`fastboot oem unlock` just reboots in recovery.

This setup was able to perform fastboot flashing on AOSP Extended and LineageOS (tested not even week ago).

I've also followed instructions from https://divestos.org/index.php?page=bootloader#fastboot without success.

I tried to do `adb reboot bootloader` prior to the flashing from an authorized system without success

## Hypothesis
DivestOS failed to unlock the bootloader

**Comment by SkewedZeppelin** [comment]
You were locked before you flashed DivestOS?

**Comment by SkewedZeppelin** [comment]
I will have to sign a stock recovery for you to flash then unlock.
It is likely the same as bacon where unlock is broken with AOSP recovery.

**Comment by Kreyren** [comment]
> You were locked before you flashed DivestOS? @SkewedZeppelin 

I wasn't

> I will have to sign a stock recovery for you to flash then unlock. @SkewedZeppelin 

Thanks!!

**Comment by SkewedZeppelin** [comment]
Did you receive the file I emailed you?

**Comment by Kreyren** [comment]
Yes i got an e-mail, but currently i can perform root and change recovery and system from that TWRP so no need to unlock the bootloader (that i prefer locked).

I can do testing for you if it helps to improve DivestOS though.

**Comment by SkewedZeppelin** [comment]
OK.
I just want to minimize a scenario where you're locked out of your
device.

**Comment by Kreyren** [comment]
That's probably good idea assuming testing in https://github.com/Divested-Mobile/DivestOS-Build/issues/9

Ok will unlock it.

**Comment by SkewedZeppelin** [comment]
Oh you don't need test it.
I am sure you can do it by going by to stock.
But that is a tedious process.

**Comment by Kreyren** [comment]
I am concerned about corrupted recovery with unbootable system during the testing of #9 tbh assuming that DivestOS is touching that recovery 

So to avoid soldering the chip in a breakout board for a debugger to flash new data assuming that being a good idea on OnePlus2 since i never done it there O.o

**Comment by SkewedZeppelin** [comment]
That is a real possibility.
I am unsure why TWRP fails to flash the DivestOS ZIPs.

**Comment by Kreyren** [comment]
>  There are also easier ways to restore that device iirc.

Elaborate?

**Comment by SkewedZeppelin** [comment]
> Elaborate

https://forums.oneplus.com/threads/updated-28-06-2016-mega-unbrick-guide-for-a-hard-bricked-oneplus-2.347607/

Method 1 of bulk mode + firehose can be trivially used to restore partitions.

It is why firehose files are so hard to get for newer devices and why verified boot is so important.

A locked vs unlocked device is no difference when a signed firehose is available for it.

**Comment by Kreyren** [comment]
Ah ok O.o
